src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +231,23 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * @param type       The type of logs being requested.
+ * @param timeout    The timeout for getting the logs.  If the timeout expires, completion
+ *                   will be called with whatever logs have been retrieved by that point
+ *                   (which might be none, and might include partial logs).
+ *                   If the timeout is null, logs will be retreived when processing is
+ *                   complete and completion is called.
```

Comment:
```suggestion
 *                   If the timeout is nil, it will never expire and completion will not be called until
 *                   the logs are fully retrieved.
```